packages/core-web/src/index.js

```diff
@@ -89,6 +89,31 @@ window.handleSiteNavClick = function (elem, useAnchor = true) {
   dropdownIcon.classList.toggle('site-nav-rotate-icon');
 };
 
+window.handleScrollTop = function () {
+  document.body.scrollIntoView({ block: 'start', behavior: 'smooth' });
+};
+
+function displayScrollTopButton() {
+  const scrollTopButton = document.querySelector('#scroll-top-button');
+  if (document.body.scrollTop > 20 || document.documentElement.scrollTop > 20) {
+    scrollTopButton.style.display = 'block';
+  } else {
+    scrollTopButton.style.display = 'none';
+  }
+}
+
+function initDisplayScrollTopButton() {
+  let timer;
+  window.addEventListener('scroll', () => {
```

Comment:
Hi Zeyu! Thanks for reviewing my code. 

The timeout is actually to delay the call to the function displayScrollTopButton(), meaning that the call to the function will only be made if user has stopped scrolling for more than 100ms. I think this would give significant improvement to the performance but correct me if I'm wrong (quite inexperienced in performance-related issues). 

I think that your idea "maybe we can introduce a timeout to hide the button after 1/2s instead? (to avoid obscuring the content)" is really good. I'll look into this!

Reply:
> The timeout is actually to delay the call to the function displayScrollTopButton()

Ahh ok; You can refer to [interaction observers](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API) to set up the scroll-button-appears-when-not-at-the-top behaviour in `displayScrollTopButton` if you're really keen on optimizing the performance, it may also end up being easier /clearer to implement.

But I'm fine with either since performance after the initial render isn't really an issue for a static site : -)

Reply:
Hi Zeyu, I tested out your idea of "hiding the button after 1/2s" but I found it to be difficult for users to be able to use the "scroll-to-top" button. 

Essentially, I feel like they have to "chase" for the button before it disappears within the timeframe (if the timeout is too short). However, if the timeout is too long, the button will obscure the content on the user's screen. 

I feel that it is hard to decide on an appropriate timeout value, such that it strikes a balance. Thus, I decided to experiment with "lightening" the button instead. What I have in mind is that we will lighten the button to a point where it wouldn't quite obscure the user's screen. 

so, currently there are 4 states of opacity: 15% (lighten), 40% (normal), 70% (hover), 100% (active)

Whenever the user scrolls, the button will turn to 40% normal opacity. But when user stops scrolling, the button will turn to 15% lightened opacity after 1 second. 

I tested the above out and I think it works pretty well for me, but I would love to hear your opinion!

Reply:
> Essentially, I feel like they have to "chase" for the button before it disappears within the timeframe

hadn't thought of that, that makes sense.